### ISSUE TITLE
Modal closing behavior on 'accept' button clicked

### ISSUE THREAD

**Issue by sachinkantipudi** [PredixDev/px-modal#34]
Hi,

## Expected behavior and actual behavior:
Expected: When positive action button is disabled (say form empty) in the modal, clicking on it should not close the modal
Actual: Modal is closing

## Steps to reproduce the problem:
Have a modal with custom header and body (div slot). Initial state of the positive action button is disabled (let's say), and now click on positive action button. Currently modal is closing. It should close only when you click on enabled button. 

## Environment (_component version number, Browser (including version), operating system, hardware, etc_):
px-modal: 3.2.1

## Screenshots (_optional, but very helpful_):
<img width="482" alt="screen shot 2018-02-05 at 5 00 42 pm" src="https://user-images.githubusercontent.com/13139814/35836633-2de7c4fa-0a96-11e8-99f3-ff8b2d2182ac.png">

## Root cause found:
`<div class="modal__accept-trigger" on-tap="_handleAcceptTriggerTapped">`
on-tap should not be there on parent div, it should affect only button inside that div.

**Comment by davidsteinberger** [comment]
IMO this can be achieved if the button gets into a disabled state. Does that work for you:

https://codepen.io/anon/pen/JpKBaZ?editors=1000

**Comment by randyaskin** [comment]
Yes good point @davidsteinberger - if you use both class `btn--disabled` (for the styling) and the `disabled` attribute (for preventing click events) the handler will not be fired.

**Comment by davidrleonard** [comment]
There was a bug in px-buttons-design `.btn--disabled` class that was overriding the default browser behavior for buttons with the `disabled` attribute. The following button would still fire click events:

```
<button class="btn btn--disabled" disabled>Fires click events</button>
```

We fixed this in px-buttons-design [v2.1.0](https://github.com/PredixDev/px-buttons-design/releases/tag/v2.1.0). If you re-run bower install, pick up the new px-buttons-design, and re-run your SCSS build you should get the new version of the `.btn--disabled` class. Now adding the `disabled` attribute to a button tag will disable click events.

**Comment by davidsteinberger** [comment]
Ha, and I was pulling my hair out reasoning about the `pointer-events` setting. Thanks much!

**Comment by sachinkantipudi** [comment]
Thanks for fixing it David!
I noticed, with disabled property set, you don't event need `.btn--disabled` class. So, I just have the below now
`<button slot="accept-trigger" class="btn btn--call-to-action" disabled$="[[!enableSave]]">[[positiveBtnLabel]]</button>`

**Comment by davidrleonard** [comment]
Hi all. We rolled back the px-buttons-design v2.1.0 release. It caused a bunch of unintended issues for existing px-buttons-design consumers. Mainly, it actually makes buttons with `.btn--disabled` not emit click events in most circumstances. It seems this issue is more complex than I hoped.

Please re-install your bower components and make sure your range for px-buttons-design is `^2.0.0` or similar, not `^2.1.0`. I'll dig into this more and have a new px-buttons-design release out as soon as possible that provides a fix but doesn't break existing use.

**Comment by davidrleonard** [comment]
OK, I think we've got this tracked down now. I released a patch for px-modal, v3.2.3, which should address this issue. Please update to this new version, and make sure you re-installed your bower components from scratch to pick up the px-buttons-design v2.0.6, which is the latest after we rolled back the last update.

If you're interested, here's what was happening:

1. A button is passed into the accept-trigger slot of the modal. It has the classes `btn` and `btn--disabled`, and the attribute `disabled`.

```html
<px-modal header-text="Message" body-text="Lorem ipsum..." reject-text="Cancel" opened>
  <button slot="accept-trigger" class="btn btn--disabled" disabled>Accept</button>
</px-modal>
```

2. The `btn--disabled` class from px-buttons-design includes the CSS rule `pointer-events: none`
3. The user clicks on the disabled button
4. A click event is fired, but not from the button. The click event is fired from the button container, which wraps around the accept-trigger slot in the px-modal shadow root.

This is what was tripping me up. If you set `pointer-events: unset` directly on the disabled button in the example above, this click event is no longer fired from the container.

This looks like a quirk of how the browser handles pointer-events unset vs. none. When pointer-events is none, the browser allows the container that holds the thing that will no longer fire pointer events to fire its own events. When pointer-events is unset, the browser does not let the container fire any events, even though it has a height/width and something inside of it. 

I fixed this by checking to make sure the click event is coming from something other than the button container. If it is coming from the button container we bail out and don't close the modal. I added test cases to cover this as best I can. It is hard to simulate real click behaviors in tests.

**Comment by sachinkantipudi** [comment]
Just wanted to note that `pointer-events: unset` still fires click event. It just unsets `pointer-events: none`. Disabled attribute ensures click event will not be fired at all. So
`<button slot="accept-trigger" class="btn" disabled>click me</button>`
will work with no issues. Only when you add, `btn--disabled` class which has `pointer-events: none`, event is fired by the parent container and handled as it has listener.
